belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/api/InputValidationIssue.java

```diff
@@ -133,34 +143,84 @@ public List<Input<?>> getInputs() {
     }
 
     public void setInputs(List<Input<?>> inputs) {
-        verifyNoInNameValue();
+
         this.inputs.clear();
-        this.inputs.addAll(inputs);
+
+        if (inputs == null) {
+            return;
+        }
+
+        if (hasInNameValue()) {
+            throw new IllegalArgumentException(INPUTS_AND_IN_NAME_VALUE_ARE_MUTUALLY_EXCLUSIVE);
+        }
+
+        List<Input<?>> filteredInputs = inputs.stream().filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (filteredInputs.size() == 1) {
+            throw new IllegalArgumentException(INPUTS_SETTER_ONE_ITEM);
+        }
+
+        this.inputs.addAll(filteredInputs);
     }
 
     public void setInputs(Input<?>... inputs) {
-        verifyNoInNameValue();
-        this.inputs.clear();
-        this.inputs.addAll(Arrays.asList(inputs));
+        if (inputs == null) {
+            return;
+        }
+        setInputs(Arrays.asList(inputs));
+    }
+
+    private void clearInNameValue() {
+        this.in = null;
+        this.name = null;
+        this.value = null;
     }
 
     public void addInput(Input<?> input) {
-        verifyNoInNameValue();
-        inputs.add(input);
+
+        if (input == null) {
+            return;
+        }
+
+        if (hasInNameValue()) {
+            this.inputs.add(new Input<>(in, name, value));
+            this.inputs.add(input);
+            clearInNameValue();
+        } else if (!inputs.isEmpty()) {
+            this.inputs.add(input);
+        } else {
+            in(input.getIn(), input.getName(), input.getValue());
+        }
     }
 
-    private void verifyNoInputs() {
-        if (!inputs.isEmpty()) {
-            throw new IllegalArgumentException(INPUTS_AND_IN_NAME_VALUE_ARE_MUTUALLY_EXCLUSIVE);
+    public void addInputs(List<Input<?>> input) {
```

Comment:
Are we giving the user too many options?

setInputs, addInput, addInputs, input, inputs

Reply:
Yes, I would streamline them:
* inputs (list and varargs)
  * I would drop input as it suggests a single one could be enough to create a valid instance (which we'll block in this PR)
  * "These are the inputs"
* addInputs (list and varags)
  * Drop addInput for same reason as input 

I would simply drop setInputs, its use case is covered inputs

Reply:
>   * Drop addInput for same reason as input
> 

I don't know about deleting addInput. It kind of means that you want to add an input to what  alreadt exists (even if there is nothing). But adding a single input is covered by addInput(varargs) in any case

Reply:
It's hard to imagine all ways a developer would use it, but maybe it can be simply become
* inputs (collection and varargs)
  * These are the input for this issue
* addInput(Input input)
* addInputs(collection)
  * No varargs as it doesn't seem that useful, you add a single entry or multiple you've collected elsewhere (and most likely into a collection

I wouldn't even be opposed against dropping "addInputs(collection)" and simply supporting:
* inputs (collection and varargs)
* addInput(Input input)

I feel that is simple enough while still supporting all use-cases.

Reply:
As discussed during the meeting we'll go ahead with the last suggestion:
- inputs (collection and varargs) for fluent building of the issue
- addInput(Input input) for dynamically adding input, e.g. in a for-loop